packages/edit-site/src/components/editor/index.js

```diff
@@ -134,22 +134,15 @@ function Editor() {
 				const {
 					__experimentalGetTemplateInfo: getTemplateInfo,
 				} = select( 'core/editor' );
-				entitiesToSave.forEach( ( { kind, name, key } ) => {
+				entitiesToSave.forEach( ( { kind, name, key, title } ) => {
 					const record = getEditedEntityRecord( kind, name, key );
-
-					if ( 'postType' === kind && name === 'wp_template' ) {
-						const { title } = getTemplateInfo( record );
-						return editEntityRecord( kind, name, key, {
-							status: 'publish',
-							title,
-						} );
+					if ( kind === 'postType' && name === 'wp_template' ) {
+						( { title } = getTemplateInfo( record ) );
 					}
-
-					const edits = record.slug
-						? { status: 'publish', title: record.slug }
-						: { status: 'publish' };
-
-					editEntityRecord( kind, name, key, edits );
+					editEntityRecord( kind, name, key, {
+						status: 'publish',
+						title: title || record.slug,
+					} );
```

Comment:
What is this supposed to do? I mean the whole function `closeEntitiesSavedStates`? Why are we  setting status and title here?

Reply:
I think setting the `status` ensures that `auto-draft` templates are set to `publish`.

Title part is needed to give a title for `wp_template_part`. Since they don't have custom titles, they are using the `slug` as a title in every case. On the other hand we want to preserve the title if the entity has one. (Basically everything except `wp_template_part`

Reply:
So something to be done before "saving" right. I don't understand why it's in `closeEntitiesSavedStates`

Reply:
I guess the  `close` prop is misnamed. It should be something more like  `preSaveCallback`